lib/declare_schema/model/habtm_model_shim.rb

```diff
@@ -53,8 +57,7 @@ def _declared_primary_key
 
       def index_definitions_with_primary_key
         [
-          IndexDefinition.new(self, foreign_keys, unique: true, name: ::DeclareSchema::Model::IndexDefinition::PRIMARY_KEY_NAME),
-          IndexDefinition.new(self, foreign_keys.last) # not unique by itself; combines with primary key to be unique
+          IndexDefinition.new(self, foreign_keys, unique: true, name: index_name) # creates a primary composite key on both foriegn keys
```

Comment:
`index_definitions_with_primary_key` is a method used on both Model and HabtmModel. This may now read better with a new singular name like `index_definition` but also I don't want to stray too far since the migrator generator uses this accessor for both classes.

Reply:
Yea, lets leave the naming as is for now and we can clean that up later if we so choose.